src/points/PointsDiscoverCard.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleProp, StyleSheet, Text, View, ViewStyle } from 'react-native'
+import { PointsEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Touchable from 'src/components/Touchable'
+import LogoHeart from 'src/icons/LogoHeart'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { pointsBalanceSelector } from 'src/points/selectors'
+import { useSelector } from 'src/redux/hooks'
+import { getFeatureGate } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+interface Props {
+  style?: StyleProp<ViewStyle>
+}
+
+export default function PointsDiscoverCard({ style }: Props) {
+  const showPoints = getFeatureGate(StatsigFeatureGates.SHOW_POINTS)
```

Comment:
Why is the gate checked within the component? I suppose if we're using this card in multiple places then this pattern would reduce some duplication, but my general instinct would be for `<PointsDiscoverCard/>` to _always_ render the card, and to handle feature gating one level above. Not super important to me, but is there any reason in particular to prefer this pattern?

Reply:
yeah, the main reason is modularity -- to encapsulate all related logic within the component, so the parent component doesn't care about conditions. it just provides a place for this card to be rendered in.

currently, we have a single condition -- the feature gate, but it could be many conditions as well. in that case parent component should handle them all and could become kind of bloated.

also, that way we can test the component in isolation.

those overweight for me the possible performance issue caused by unconditional rendering. I think it could be ignored for this particular, relatively lightweight component.

fwiw, the neighboring [EarnCard](https://github.com/valora-inc/wallet/blob/258f77a1958bd88a92bd7efc4cf2c7517b568c48/src/earn/EarnCard.tsx#L16) is implemented in the same way.

I would prefer to keep consistency for now and re-use the `EarnCard` "modular" pattern.